src/extras/AnimatedSprite.js

```diff
@@ -30,13 +30,61 @@ import * as core from '../core';
 export default class AnimatedSprite extends core.Sprite
 {
     /**
-     * @param {PIXI.Texture[]|FrameObject[]} textures - an array of {@link PIXI.Texture} or frame
-     *  objects that make up the animation
+     * @param {PIXI.Texture[]|FrameObject[]|object} textures - an array of {@link PIXI.Texture} or frame
+     * objects that make up the animation. it also can be a series of 'named animations', an object containing
+     * arrays of textures or frames.
+     * @param {string} initialAnimation - name of the initial animation. must exists inside the textures if textures
+     * is an object containing named animations. this param must be ommited if textures is an Array.
      * @param {boolean} [autoUpdate=true] - Whether to use PIXI.ticker.shared to auto update animation time.
      */
-    constructor(textures, autoUpdate)
+    constructor(textures, initialAnimation, autoUpdate)
```

Comment:
-1 for this BC

overall code seems bloated with all these statements

I would suggest using some defined structure for `textures`

here is some example for button with skins:

![image](https://user-images.githubusercontent.com/899452/32759195-cffe70dc-c8f0-11e7-8a60-3dfc79a3b1f7.png)

Reply:
Hi, @cursedcoder!

I am trying to not break backwards compatibility, indeed it would be much simpler, but i am not there to break someone's else code. minor releases are not intended to to such thing.

Also your image suggestion uses TypeScript idioms which are simply not found on pixi.js codebase, but i am still interested to be able to offer to other people the named animation idiom. 

Have you anoy other suggestion regarding how could i introduce this new idiom? 
- it must be backwards compatible (the inheritance strategy did that, but i was suggested to merge functionality on AnimatedSprite)
- it must pass happily from "npm run lint"

I'm willing to send another patch, but it only will pass if i got someone to support and guide me on this.